### PR TITLE
[FL-2970] Untangle NFC_APP_FOLDER from nfc_device

### PR DESCRIPTION
**Pull request by Astrrra**
# What's new

- NFC_APP_FOLDER is now defined in the NFC app, not the NFC device

# Verification 

- Try all the saving/deleting/editing operations, verify that the proper directory gets created when it's not present and that the file is saved at the correct location.

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: lib/nfc/nfc_device.h [hunk 1]
```diff
@@ -105,15 +104,15 @@ bool nfc_device_load(NfcDevice* dev, const char* file_path, bool show_dialog);
 
 bool nfc_device_load_key_cache(NfcDevice* dev);
 
-bool nfc_file_select(NfcDevice* dev);
+bool nfc_file_select(NfcDevice* dev, const char* folder);
```
Comment (skotopes) [inline lib/nfc/nfc_device.h]: Looks like some kind of environment variable for NfcDevice, why not to store in NfcDevice struct?
Or create something like NfcEnvironment and pass that kind of things through it.
Check SubGhzEnvironment for examples.

**Review by skotopes — CHANGES_REQUESTED** [review]

File: lib/nfc/nfc_device.h (hunk 1 above)
Comment (Astrrra) [reply lib/nfc/nfc_device.h]: True, storing it in NfcDevice sounds like the best option, as NfcEnvironment would be kinda excessive, I think.

File: firmware/targets/f7/api_symbols.csv [hunk 2]
```diff
@@ -1,5 +1,5 @@
 entry,status,name,type,params
-Version,+,10.1,,
+Version,+,12.0,,
```
Comment (skotopes) [inline firmware/targets/f7/api_symbols.csv]: Why?

**Review by skotopes — CHANGES_REQUESTED** [review]

File: applications/main/nfc/scenes/nfc_scene_delete.c [hunk 3]
```diff
@@ -52,6 +52,7 @@ void nfc_scene_delete_on_enter(void* context) {
 
 bool nfc_scene_delete_on_event(void* context, SceneManagerEvent event) {
     Nfc* nfc = context;
+    furi_string_printf(nfc->dev->folder, "%s", NFC_APP_FOLDER);
```
Comment (skotopes) [inline applications/main/nfc/scenes/nfc_scene_delete.c]: Looks like something that can be passed to allocator.

**Review by skotopes — CHANGES_REQUESTED** [review]

File: applications/main/nfc/scenes/nfc_scene_delete.c (hunk 3 above)
Comment (Astrrra) [reply applications/main/nfc/scenes/nfc_scene_delete.c]: That's fair, fixed now

File: firmware/targets/f7/api_symbols.csv (hunk 2 above)
Comment (Astrrra) [reply firmware/targets/f7/api_symbols.csv]: Oh, didn't notice that, fixed

File: lib/nfc/nfc_device.c [hunk 4]
```diff
@@ -1018,6 +1020,11 @@ static void nfc_device_get_shadow_path(FuriString* orig_path, FuriString* shadow
     furi_string_cat_printf(shadow_path, "%s", NFC_APP_SHADOW_EXTENSION);
 }
 
+static void nfc_device_get_folder_from_path(FuriString* path, FuriString* folder) {
+    size_t last_slash = furi_string_search_rchar(path, '/');
```
Comment (skotopes) [inline lib/nfc/nfc_device.c:1024]: How about STRING_FAILURE ?

File: applications/plugins/nfc_magic/nfc_magic.c [hunk 5]
```diff
@@ -38,6 +38,7 @@ NfcMagic* nfc_magic_alloc() {
     nfc_magic->worker = nfc_magic_worker_alloc();
     nfc_magic->view_dispatcher = view_dispatcher_alloc();
     nfc_magic->scene_manager = scene_manager_alloc(&nfc_magic_scene_handlers, nfc_magic);
+    furi_string_printf(nfc_magic->nfc_dev->folder, "%s", NFC_APP_FOLDER);
```
Comment (skotopes) [inline applications/plugins/nfc_magic/nfc_magic.c]: furi_string_set ?

File: applications/main/nfc/nfc.c [hunk 6]
```diff
@@ -39,6 +39,7 @@ Nfc* nfc_alloc() {
     nfc->worker = nfc_worker_alloc();
     nfc->view_dispatcher = view_dispatcher_alloc();
     nfc->scene_manager = scene_manager_alloc(&nfc_scene_handlers, nfc);
+    furi_string_printf(nfc->dev->folder, "%s", NFC_APP_FOLDER);
```
Comment (skotopes) [inline applications/main/nfc/nfc.c]: furi_string_set ?

File: lib/nfc/nfc_device.c [hunk 7]
```diff
@@ -1028,8 +1035,18 @@ bool nfc_device_save(NfcDevice* dev, const char* dev_name) {
     temp_str = furi_string_alloc();
 
     do {
-        // Create nfc directory if necessary
-        if(!storage_simply_mkdir(dev->storage, NFC_APP_FOLDER)) break;
+        // Create directory if necessary
+        FuriString* folder = furi_string_alloc();
+        // Get folder from filename (filename is in the form of "folder/filename.nfc", so the folder is "folder/")
+        furi_string_printf(temp_str, "%s", dev_name);
```
Comment (skotopes) [inline lib/nfc/nfc_device.c]: furi_string_set ?

**Review by skotopes — CHANGES_REQUESTED** [review]

**Review by skotopes — APPROVED** [review]

**Pull request merged**